packages/node/src/integrations/tracing/nest/nest.ts

```diff
@@ -87,11 +87,16 @@ export function setupNestErrorHandler(app: MinimalNestJsApp, baseFilter: NestJsE
         const originalCatch = Reflect.get(target, prop, receiver);
 
         return (exception: unknown, host: unknown) => {
-          const status_code = (exception as ExpectedException).status;
-          const error_property = (exception as ExpectedException).error;
-
-          // don't report expected errors
-          if (status_code !== undefined || error_property !== undefined) {
+          const exceptionIsObject = typeof exception === 'object' && exception !== null;
+          const exceptionStatusCode = exceptionIsObject && 'status' in exception ? exception.status : null;
+          const exceptionErrorProperty = exceptionIsObject && 'error' in exception ? exception.error : null;
+
+          /*
+          Don't report expected NestJS control flow errors
+          - `HttpException` errors will have a `status` property
+          - `RpcException` errors will have an `error` property
+           */
+          if (exceptionStatusCode !== null || exceptionErrorProperty !== null) {
```

Comment:
l: I would flip the condition: We report the error if this and this condition is true. To me the early return is a weird pattern here but I'll leave it up to you.